gazebo_ros/scripts/spawn_model.py

```diff
@@ -0,0 +1,312 @@
+                            help='change ROS namespace of gazebo-plugins')
+        parser.add_argument('-unpause', action='store_true',
+                            help='unpause physics after spawning model')
+        parser.add_argument('-wait', type=str, metavar='MODEL_NAME',
+                            help='Wait for model to exist')
+        parser.add_argument('-x', type=float, default=0,
+                            help='x component of initial position, meters')
+        parser.add_argument('-y', type=float, default=0,
+                            help='y component of initial position, meters')
+        parser.add_argument('-z', type=float, default=0,
+                            help='z component of initial position, meters')
+        parser.add_argument('-R', type=float, default=0,
+                            help='roll angle of initial orientation, radians')
+        parser.add_argument('-P', type=float, default=0,
+                            help='pitch angle of initial orientation, radians')
+        parser.add_argument('-Y', type=float, default=0,
+                            help='yaw angle of initial orientation, radians')
+
+        # TODO(shivesh): Wait for /set_model_configuration
+        # (https://github.com/ros-simulation/gazebo_ros_pkgs/issues/779)
+        # parser.add_argument('-J', dest='joints', default=[], action='append',
+        #     metavar=('JOINT_NAME', 'JOINT_POSITION'), type=str, nargs=2,
+        #     help='initialize the specified joint at the specified position')
+
+        parser.add_argument('-package_to_model', action='store_true', help='convert urdf \
+                            <mesh filename="package://..." to <mesh filename="model://..."')
+
+        # TODO(shivesh): Wait for https://github.com/ros2/rclpy/issues/244
+        # parser.add_argument('-b', dest='bond', action='store_true', help='bond to gazebo \
+        #                      and delete the model when this program is interrupted')
+        self.args = parser.parse_args(args[1:])
+
+        # TODO(shivesh): Wait for /set_model_configuration
+        # (https://github.com/ros-simulation/gazebo_ros_pkgs/issues/779)
+        # Convert position of joints to floats
+        # for i in range(len(self.args.joints)):
+        #     self.args.joints[i][1] = float(self.args.joints[i][1])
+
+    def run(self):
+        """
+        Run node, spawning model and doing other actions as configured in program arguments.
+
+        Returns exit code, 1 for failure, 0 for success
+        """
+        # Wait for model to exist if wait flag is enabled
+        if self.args.wait:
+            self.model_exists = False
+
+            def models_cb(models):
+                self.model_exists = self.args.wait in models.name
+
+            self.subscription = self.create_subscription(
+                ModelStates, '%s/model_states' % self.args.gazebo_namespace, models_cb, 10)
```

Comment:
This topic doesn't exist in ROS2 yet, we could add it to `gazebo_ros_state`

Reply:
I have commented it for now. Will get back to it after porting model states